### DISCUSSION TITLE
Executing codes in Kin's online editor

### DISCUSSION THREAD

**Discussion by irfiacre** [kin-lang/kin#179] [Q&A] [answered]
Hello,

I noticed that the `Online-editor` does not execute added code content (the clear button also does not work). In my quest to fix this, I run into a blocker. It seems like the [kin package](https://www.npmjs.com/package/@kin-lang/kin) is not compatible with the web because it uses the `child_process` that is not compatible with the browser.

Please let me know if there is a possible alternative that can be used or if you have a suggestion feel free to share them!

**Comment by regisrex** [comment]
I have this kind of crazy idea, I don't know if it can work, but.. we can try.
Whatif we make something like a websocket api that handles the communication between the web interface and the Kin REPL. So that whenever someone writes code and clicks on run, we will send the code to the server with kin installed, then stream the server logs. How about that ?

**Reply by pacifiquem** [reply]
A very good suggestion @regisrex, Been exproling other options like making use of `Next.js' API Routes` as this feature can help us to access Node.js' built-in modules but in the end we'll need to stream server logs! Let's approach this problem like this.

**Comment by pacifiquem** [comment]
@irfiacre do you want to contribute to this? Or you were reporting issue to the team?

**Reply by irfiacre** [reply]
Yes I am interested in contributing.

**Comment by pacifiquem** [comment, marked as answer]
Thanks @irfiacre ,  I'm going to open an issue for this and you'll open a PR for fixing it, here's how it'll be done.
- We're have an external express server which will help us to run Kin
- We'll be using Sockets to connect to the server
- When a user clicks run code button, we'll connect to server via websockets and run code user sent
- We'll be streaming logs from the server to frontend in real-time.

**Reply by pacifiquem** [reply]
@irfiacre check issue kin-lang/editor#3

**Reply by irfiacre** [reply]
Thanks, how can one access the express server?

**Reply by pacifiquem** [reply]
Setup one @irfiacre we'll take care of production environment setup!

**Reply by irfiacre** [reply]
Hello,

When trying to push, I get the following error:
```
Please make sure you have the correct access rights
and the repository exists.
```

Additionally trying to create a repo is restricted!

**Reply by pacifiquem** [reply]
Please fork the repo and open a pull request from the fork, it'll work @irfiacre
